files/boot.py

```diff
@@ -1,85 +1,114 @@
 #!/usr/bin/python
 """ Prints an iPXE script to stdout to start a kickstart & more """
-import syslog
+from __future__ import print_function
 import sys
 import os
 import socket
+import syslog
+import json
+
 sys.stderr = sys.stdout
-print "Content-Type: text/plain"
-print
+print("Content-Type: text/plain")
+print()
+
+PXE_HEADER = "#!ipxe"
+if "gPXE" in os.environ["HTTP_USER_AGENT"]:
+    PXE_HEADER = "#!gpxe"
+
+
+def pxe_abort():
+    """Abort the PXE boot, continue with the next boot device in the BIOS boot order"""
+    print(PXE_HEADER)
+    print("exit")
+    sys.exit(0)
 
 syslog.openlog("boot.py")
 syslog.setlogmask(syslog.LOG_UPTO(syslog.LOG_INFO))
 
-# from the name, e.g. c1-3.cloud.example.org take c1-3
-hostname = socket.gethostbyaddr(os.environ["REMOTE_ADDR"])[0].split(".")[0]
-syslog.syslog(syslog.LOG_DEBUG, "Got boot iPXE request from " + hostname)
+# from the name, e.g. c1-3.cloud.example.org take c1-3 and put it in HOSTNAME
+try:
+    FQDN = socket.gethostbyaddr(os.environ["REMOTE_ADDR"])[0]
+    HOSTNAME = FQDN.split(".")[0]
+except Exception as e_xcep:
+    syslog.syslog(
+        syslog.LOG_ERR,
+        str(e_xcep) + " HOSTNAME wasn't found in /var/www/provision/nodes/pxe_nodes.json",
+    )
+    pxe_abort()
+
+syslog.syslog(syslog.LOG_DEBUG, "Got boot iPXE request from " + HOSTNAME + "(" + FQDN + ")")
 
-started = False
-# When a hypervisor restarts without a reinstall/memtest it is expected that the tries fails. 
+####### Memtest
+STARTED = False
+# When a hypervisor restarts without a reinstall/memtest it is expected that the tries fails.
 try:
-    os.stat("/var/www/provision/memtest86/" + hostname)
-    os.remove("/var/www/provision/memtest86/" + hostname)
-    f = open("/var/www/provision/nodes/" + hostname + ".conf")
-
-    syslog.syslog(syslog.LOG_INFO, "Memtesting node " + hostname)
-    nodesettings = {}
-    for line in f.readlines():
-      #for every line, e.g. "key=value", set nodesettings["key"]="value"
-      #comment lines will throw an error, skip them
-        try:
-            nodesettings[line.split("=")[0]] = line.split("=", 1)[1].strip()
-        except:
-            pass
-
-    f.close()
+    # for convenience create a file called HOSTNAME == short hostname
```

Comment:
Is this a legacy comment? No files are being created in the following lines.

Reply:
The "for convenience.." bit?
The files in memtest86/ (or reinstall/ ) directory are not created as part of this script and from what I remember have never been. An admin(or tool) would touch the files and then boot.py removes them if a matching node http gets boot.py

Reply:
Basically I read this as

`# for convenience [as a python script, I shall] create a file called HOSTNAME == short hostname`

instead of 

`# for convenience [users only need to] create a file called HOSTNAME == short hostname`

but that's probably just my asperger flaring. Nevermind :)

Reply:
No bother. Stay tuned for some new and improved commentary track.